staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -58,14 +57,6 @@ const (
 
 var (
 	defaultCacheDir = filepath.Join(homedir.HomeDir(), ".kube", "http-cache")
-
-	ErrEmptyConfig = errors.New(`Missing or incomplete configuration info.  Please point to an existing, complete config file:
```

Comment:
moving this type is not a change we could pick back to 1.18... I'd suggest something like this instead:

```diff
diff --git a/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go b/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go
index 5ded3e98ba8..4a6fb5c3606 100644
--- a/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go
+++ b/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go
@@ -59,13 +59,13 @@ const (
 var (
 	defaultCacheDir = filepath.Join(homedir.HomeDir(), ".kube", "http-cache")
 
-	ErrEmptyConfig = errors.New(`Missing or incomplete configuration info.  Please point to an existing, complete config file:
+	ErrEmptyConfig = clientcmd.NewEmptyConfigError(errors.New(`Missing or incomplete configuration info.  Please point to an existing, complete config file:
 
   1. Via the command-line flag --kubeconfig
   2. Via the KUBECONFIG environment variable
   3. In your home directory as ~/.kube/config
 
-To view or setup config directly use the 'config' command.`)
+To view or setup config directly use the 'config' command.`))
 )
 
 // RESTClientGetter is an interface that the ConfigFlags describe to provide an easier way to mock for commands
diff --git a/staging/src/k8s.io/client-go/tools/clientcmd/validation.go b/staging/src/k8s.io/client-go/tools/clientcmd/validation.go
index afe6f80b39e..cf85c0a4acc 100644
--- a/staging/src/k8s.io/client-go/tools/clientcmd/validation.go
+++ b/staging/src/k8s.io/client-go/tools/clientcmd/validation.go
@@ -30,11 +30,20 @@ import (
 
 var (
 	ErrNoContext   = errors.New("no context chosen")
-	ErrEmptyConfig = errors.New("no configuration has been provided, try setting KUBERNETES_MASTER environment variable")
+	ErrEmptyConfig = NewEmptyConfigError(errors.New("no configuration has been provided, try setting KUBERNETES_MASTER environment variable"))
 	// message is for consistency with old behavior
 	ErrEmptyCluster = errors.New("cluster has no server defined")
 )
 
+// NewEmptyConfigError returns an error wrapping the given error which IsEmptyConfig() will recognize as an empty config error
+func NewEmptyConfigError(err error) error {
+	return emptyConfigError{err}
+}
+
+type emptyConfigError struct {
+	error
+}
+
 type errContextNotFound struct {
 	ContextName string
 }
@@ -60,9 +69,14 @@ func IsContextNotFound(err error) bool {
 func IsEmptyConfig(err error) bool {
 	switch t := err.(type) {
 	case errConfigurationInvalid:
-		return len(t) == 1 && t[0] == ErrEmptyConfig
+		if len(t) != 1 {
+			return false
+		}
+		_, ok := t[0].(emptyConfigError)
+		return ok
 	}
-	return err == ErrEmptyConfig
+	_, ok := err.(emptyConfigError)
+	return ok
 }
 
 // errConfigurationInvalid is a set of errors indicating the configuration is invalid.
```